doc/guide/getting-started.md

```diff
@@ -0,0 +1,179 @@
+EOF
+
+exit
+```
+
+To allow openFPGAloader to program our device, add the following rules:
+```sh
+sudo su
+cat <<EOF > /etc/udev/rules.d/99-openfpgaloader.rules
+# Copy this file to /etc/udev/rules.d/
+
+ACTION!="add|change", GOTO="openfpgaloader_rules_end"
+
+# gpiochip subsystem
+SUBSYSTEM=="gpio", MODE="0664", GROUP="plugdev", TAG+="uaccess"
+
+SUBSYSTEM!="usb|tty|hidraw", GOTO="openfpgaloader_rules_end"
+
+# Original FT232/FT245 VID:PID
+ATTRS{idVendor}=="0403", ATTRS{idProduct}=="6001", MODE="664", GROUP="plugdev", TAG+="uaccess"
+
+# Original FT2232 VID:PID
+ATTRS{idVendor}=="0403", ATTRS{idProduct}=="6010", MODE="664", GROUP="plugdev", TAG+="uaccess"
+
+# Original FT4232 VID:PID
+ATTRS{idVendor}=="0403", ATTRS{idProduct}=="6011", MODE="664", GROUP="plugdev", TAG+="uaccess"
+
+# Original FT232H VID:PID
+ATTRS{idVendor}=="0403", ATTRS{idProduct}=="6014", MODE="664", GROUP="plugdev", TAG+="uaccess"
+
+# Original FT231X VID:PID
+ATTRS{idVendor}=="0403", ATTRS{idProduct}=="6015", MODE="664", GROUP="plugdev", TAG+="uaccess"
+
+# anlogic cable
+ATTRS{idVendor}=="0547", ATTRS{idProduct}=="1002", MODE="664", GROUP="plugdev", TAG+="uaccess"
+
+# altera usb-blaster
+ATTRS{idVendor}=="09fb", ATTRS{idProduct}=="6001", MODE="664", GROUP="plugdev", TAG+="uaccess"
+ATTRS{idVendor}=="09fb", ATTRS{idProduct}=="6002", MODE="664", GROUP="plugdev", TAG+="uaccess"
+ATTRS{idVendor}=="09fb", ATTRS{idProduct}=="6003", MODE="664", GROUP="plugdev", TAG+="uaccess"
+
+# altera usb-blasterII - uninitialized
+ATTRS{idVendor}=="09fb", ATTRS{idProduct}=="6810", MODE="664", GROUP="plugdev", TAG+="uaccess"
+# altera usb-blasterII - initialized
+ATTRS{idVendor}=="09fb", ATTRS{idProduct}=="6010", MODE="664", GROUP="plugdev", TAG+="uaccess"
+
+# dirtyJTAG
+ATTRS{idVendor}=="1209", ATTRS{idProduct}=="c0ca", MODE="664", GROUP="plugdev", TAG+="uaccess"
+
+# Jlink
+ATTRS{idVendor}=="1366", ATTRS{idProduct}=="0105", MODE="664", GROUP="plugdev", TAG+="uaccess"
+
+# NXP LPC-Link2
+ATTRS{idVendor}=="1fc9", ATTRS{idProduct}=="0090", MODE="664", GROUP="plugdev", TAG+="uaccess"
+
+# NXP ARM mbed
+ATTRS{idVendor}=="0d28", ATTRS{idProduct}=="0204", MODE="664", GROUP="plugdev", TAG+="uaccess"
+
+# icebreaker bitsy
+ATTRS{idVendor}=="1d50", ATTRS{idProduct}=="6146", MODE="664", GROUP="plugdev", TAG+="uaccess"
+
+# orbtrace-mini dfu
+ATTRS{idVendor}=="1209", ATTRS{idProduct}=="3442", MODE="664", GROUP="plugdev", TAG+="uaccess"
+
+LABEL="openfpgaloader_rules_end"
+
+EOF
+
+exit
+
+```
+
+Run the following to reload the rules:
+```sh
+sudo udevadm control --reload-rules
+sudo udevadm trigger
+```
+
+Add user to plugdev group:
+```sh
+sudo usermod -a $USER -G plugdev
+```
+
+### Bitstream
+
+Build the bitstream:
+```sh
+fusesoc --cores-root=. run --target=synth --setup --build lowrisc:sonata:system
+```
+
+### Programming
+
+Programming the FPGA:
+```sh
+openFPGALoader -b arty_a7_35t build/lowrisc_sonata_system_0/synth-vivado/lowrisc_sonata_system_0.bit
```

Comment:
Before the getting-started guide goes public, it needs to mention building the software; presently if someone has not built the hello world 'demo' application, there is no vmem file. Vivado builds a FPGA image that contains no code, hiding away the "CRITICAL WARNING" in its log file and leaving the Ibex with nothing to run; a silent FPGA board with no indication of why.

Reply:
There is a "Software" section which has these instructions. I've put a line here linking to that section.